### DISCUSSION TITLE
When importing a Bitwarden json or csv export, Vaultwarden fails to import Identity entries

### DISCUSSION THREAD

**Discussion by jasonpearce** [dani-garcia/vaultwarden#4970] [Q&A]
### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.31.0
* Web-vault version: v2024.5.1b
* OS/Arch: linux/x86_64
* Running within a container: true (Base: Alpine)
* Environment settings overridden: true
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Browser/Server Time Check: true
* Server/NTP Time Check: true
* Domain Configuration Check: false
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.46.0
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** DOMAIN, PASSWORD_ITERATIONS, ADMIN_TOKEN

```json
{
  "_duo_akey": null,
  "_enable_duo": true,
  "_enable_email_2fa": false,
  "_enable_smtp": false,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "_smtp_img_src": "cid:",
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_session_lifetime": 20,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "auth_request_purge_schedule": "30 * * * * *",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "***************",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://**************************************************************",
  "domain_origin": "*****://**************************************************************",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_2fa_auto_fallback": false,
  "email_2fa_enforce_on_verified_invite": false,
  "email_attempts_limit": 3,
  "email_change_allowed": true,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 3 * * * *",
  "emergency_request_timeout_schedule": "0 7 * * * *",
  "enable_db_wal": true,
  "enable_websocket": true,
  "event_cleanup_schedule": "0 10 0 * * *",
  "events_days_retain": null,
  "experimental_client_feature_flags": "fido2-vault-credentials",
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "invitation_expiration_hours": 120,
  "invitation_org_name": "Vaultwarden",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "Info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "",
  "org_events_enabled": false,
  "org_groups_enabled": false,
  "password_hints_allowed": true,
  "password_iterations": 8000000,
  "push_enabled": false,
  "push_identity_uri": "https://identity.bitwarden.com",
  "push_installation_id": "***",
  "push_installation_key": "***",
  "push_relay_uri": "https://push.bitwarden.com",
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sendmail_command": null,
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": true,
  "signups_domains_whitelist": "***************,**************,**********,**********************",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_embed_images": true,
  "smtp_explicit_tls": null,
  "smtp_from": "",
  "smtp_from_name": "Vaultwarden",
  "smtp_host": null,
  "smtp_password": null,
  "smtp_port": 587,
  "smtp_security": "starttls",
  "smtp_ssl": null,
  "smtp_timeout": 15,
  "smtp_username": null,
  "templates_folder": "data/templates",
  "tmp_folder": "data/tmp",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_sendmail": false,
  "use_syslog": false,
  "user_attachment_limit": null,
  "user_send_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "yubico_client_id": "***",
  "yubico_secret_key": "***",
  "yubico_server": null
}
```
</details>

### Vaultwarden Build Version

v1.31.0

### Deployment method

Other method

### Custom deployment method

Brand new Start9 StartOS virtual machine. Used their Marketplace to install Vaultwarden via the GUI.

### Reverse Proxy

No reverse proxy

### Host/Server Operating System

Linux

### Operating System Version

StartOS v0.3.5~1

### Clients

Web Vault

### Client Version

Latest version of Microsoft Edge and Firefox TOR browser

### Steps To Reproduce

1. Go to a Bitwarden vault in a browser that has Login, Card, Identity, and Secure Note objects.
2. Export vault to both .json and .csv formats
3. View both files in Notepad to verify Identity objects were exported by Bitwarden (they were) and then close without making changes or saving
4. Go to a new Vaultwarden vault in a browser
5. Create a New Organization like "2024-09-19 Bitwarden backup"
6. Select the newly created organization in Vaultwarden
7. Select Tools > Import Data
8. Import either the .json or .csv export from Bitwarden into the new Vaultwarden and new Collection (problem occurs with either)
9. Vaultwarden will report that A Logins, B Cards, C Identities, and D Secure Notes were imported
10. Browse Identity or Card and observe "There are no items to list". Both Login and Secure Note, however, have imported objects.

### Expected Result

Vaultwarden reports that both Card and Identity objects were imported. But when you browse or search for Card or Identity values, they do not appear to be present.

My expected result is that all values from the Bitwarden-created .json and .csv exports would be imported, including Card and Identity.

### Actual Result

After importing a Bitwarden-created .json or .csv Export (that does contain values for both Card and Identity that I can view in Notepad prior to importing into Vaultwarden), both Card and Identity report "There are no items to list"

### Logs

```text
I'm sorry. I could not easily find logs.
```

### Additional Context

No additional information.

**Comment by BlackDex** [comment]
Vaultwarden does not handle the import it self in the sense of converting. The clients proces, encrypt and send the imported data to the server. So if this is a big it's not something we can fix. 

Further, you are not using the latest version of Vaultwarden, and thus also not the latest web-vault we support which might have this resolved already.

**Reply by jasonpearce** [reply]
Thank you for the quick response. 

- I’m using the latest version available in the Start9 marketplace, which I think is one version behind
- If it is not something you can fix but a version update could fix it, could this be tested?

As I wait for the Start9 App Store to update to the latest version of Vaultwarden, perhaps I’ll attempt to Import into a new instance of Bitwarden and report my outcome.

**Reply by BlackDex** [reply]
You can report here, but it will not matter that much. We still can't resolve it if it doesn't work.

Also, Bitwarden it self is several versions ahead of us regarding the web-vault.
